packages/ag-charts-community/src/chart/series/shapeUtil.ts

```diff
@@ -16,44 +17,88 @@ export interface ShapeFillBBox {
 
 export function getShapeFill(
     fill: InternalAgColorType,
-    defaults: Required<InternalAgGradientColor>
+    defaultGradient: Required<InternalAgGradientColor>,
+    defaultPattern: Required<AgPatternColor>
 ): Required<InternalAgColorType>;
 export function getShapeFill(
     fill: InternalAgColorType | undefined,
-    defaults: Required<InternalAgGradientColor>
+    defaultGradient: Required<InternalAgGradientColor>,
+    defaultPattern: Required<AgPatternColor>
 ): Required<InternalAgColorType> | undefined;
 export function getShapeFill(
     fill: InternalAgColorType | undefined,
-    defaults: Required<InternalAgGradientColor>
+    defaultGradient: Required<InternalAgGradientColor>,
+    defaultPattern: Required<AgPatternColor>
 ): Required<InternalAgColorType> | undefined {
-    if (!isGradientFill(fill)) return fill as any;
+    if (isGradientFill(fill)) {
+        return {
+            type: 'gradient',
+            gradient: fill.gradient ?? defaultGradient.gradient,
+            colorStops: fill.colorStops ?? defaultGradient.colorStops,
+            bounds: fill.bounds ?? defaultGradient.bounds,
+            rotation: fill.rotation ?? defaultGradient.rotation,
+            reverse: fill.reverse ?? defaultGradient.reverse,
+        };
+    }
 
-    return {
-        type: 'gradient',
-        gradient: fill.gradient ?? defaults.gradient,
-        colorStops: fill.colorStops ?? defaults.colorStops,
-        bounds: fill.bounds ?? defaults.bounds,
-        rotation: fill.rotation ?? defaults.rotation,
-        reverse: fill.reverse ?? defaults.reverse,
-    };
+    if (isPatternFill(fill) && defaultPattern) {
+        // TODO: move this logic to theme operations
+        const pattern = fill.pattern ?? defaultPattern.pattern;
+
+        let strokeWidth = fill.strokeWidth;
+        if (
+            pattern === 'backward-slanted-lines' ||
+            pattern === 'forward-slanted-lines' ||
+            pattern === 'horizontal-lines' ||
+            pattern === 'vertical-lines'
+        ) {
+            strokeWidth ??= 8;
+        } else {
+            strokeWidth ??= 0;
+        }
+
+        const width = fill.width ?? fill.height ?? 26;
+        const height = fill.height ?? fill.width ?? 26;
+
+        return {
+            type: 'pattern',
+            pattern,
+            width,
+            height,
+            padding: fill.padding ?? defaultPattern.padding,
+            fill: fill.fill ?? defaultPattern.fill,
+            fillOpacity: fill.fillOpacity ?? defaultPattern.fillOpacity,
+            backgroundFill: fill.backgroundFill ?? defaultPattern.backgroundFill,
+            backgroundFillOpacity: fill.backgroundFillOpacity ?? defaultPattern.backgroundFillOpacity,
+            stroke: fill.stroke ?? defaultPattern.stroke,
+            strokeOpacity: fill.strokeOpacity ?? defaultPattern.strokeOpacity,
+            strokeWidth,
```

Comment:
does the user provided strokeWidth still override our defaults?

Reply:
Yep, the `??=` assignments above will ensure it takes the `fill.strokeWidth` if set by the user.

Reply:
oh yeah :)